### ISSUE TITLE
🚀 Material for MkDocs 5.0.0 RC 4

### ISSUE THREAD

**Issue by squidfunk** [squidfunk/mkdocs-material#1498]
__Help test Material 5 RC 4! [Deploy preview](https://deploy-preview-1486--mkdocs-material-preview.netlify.com/)__

Please post any problems you encounter during migration in this issue.

## Installation

Using `pip`:

```
pip install "mkdocs-material>=5.0.0rc4"
```

Using `docker`:

```
docker pull squidfunk/mkdocs-material:5.0.0rc4
```

## Features

- [x] Reactive architecture – try `__material.dialog$.next("Hi!")` in the console!
- [x] Instant loading – make Material behave like a Single Page Application!
- [x] Improved CSS customization with [CSS variables](https://github.com/squidfunk/mkdocs-material/blob/081d540127c41703da3d73aa69eb521615c1cb89/src/assets/stylesheets/base/_colors.scss#L27-L55) – define your CI colors!
- [x] Improved CSS resilience, e.g. proper sidebar locking for customized headers
- [x] Improved icon integration and configuration – now including over 5k icons!
- [x] Added possibility to use any icon for logo, repository and social links
- [x] Search UI does not freeze anymore (moved to web worker)
- [x] Search index built only once when using instant loading
- [x] Improved extensible keyboard handling
- [x] Support for [prebuilt search indexes](https://www.mkdocs.org/user-guide/configuration/#prebuild_index)
- [x] Support for displaying stars and forks for GitLab repositories
- [x] Support for scroll snapping of sidebars and search results
- [x] Reduced HTML and CSS footprint due to deprecation of Internet Explorer support
- [x] Slight facelifting of some UI elements (Admonitions, tables, ...)

## Fixed in RC 2

- [x] #1505: Search does not close – c0abaf5
- [x] #1503: Exception when setting only logo icon – fcbd47c
- [x] #1502: Tabbed content alignment – 4d386df, 447d409
- [x] #1499, #1501: Code block styling issues with `pymdownx.inline` and `inline`

## Fixed in RC 3

- [x] #1515: Return link of footnote not show after convert to pdf enhancement
- [x] Fixed Admonitions for print media
- [x] Fixed Details for Safari (iOS and macOS) 
- [x] Fixed hover states for nested items in mobile navigation
- [x] Improved rendering performance
- [x] Improved social icons and copyright notice alignment
- [x] Improved accessibility

## Fixed in RC 4

- [x] Fixed #1544: Move logo into partial for easier overriding enhancement – c9b2c1e
- [x] Fixed #1518: Allow configuration of default search query pre-processing function enhancement – 64caf62
- [x] Fixed #1507: Instant loading scroll restoration bug – 4d370fe
- [x] Fixed #1451: Nested PDFs and SVGs seem to be ignored as page content bug – 42524ae1
- [x] Fixed replacement of skip link and announcement bar on instant load – 908d34b8
- [x] Fixed bug where a `popstate` event triggered `history.pushState` again
- [x] Fixed header ellipsis when title equals site name
- [x] Fixed hover states of search input for black and white palette
- [x] Removed `required` attribute on search input
- [x] Improved global keyboard events to only emit when not in editable element
- [x] Improved color customization of details arrow icon
- [x] Improved copy-to-clipboard button sizing in Admonitions

## Migration

See the [migration guide in the deploy preview](https://deploy-preview-1486--mkdocs-material-preview.netlify.com/releases/5/).

**Comment by fkorotkov** [comment]
@squidfunk will you publish an RC Docker image as well?

**Comment by squidfunk** [comment]
@fkorotkov just pushed the docker image, see OP.

**Comment by SM-26** [comment]
Hey, I'm having some issues with custom logos/icons.

If I want to use my own PNG/SVG, where should I put it in?

this is my mkdocs.yml:
``` mkdocs.yml
theme:
  name: material
  custom_dir: 'theme'
  language: en
  feature:
    - instant
    - tabs
  palette:
    primary: 'blue grey'
    accent: 'red'
  # font:
    # text: 'Roboto'
    # code: 'Ubuntu Mono'
  icon:
    logo: "assets/images/android-chrome-256x256.png"
  favicon: "assets/images/safari-pinned-tab.svg"
```

and in my directory looks like this:
```
docs
- index.md
- robots.txt
-sitemap.xml
theme
- assets
-- javascripts
-- stylesheets
- icons
-- assets
--- images
---- android-chrome-256x256.png
---- android-chrome-256x256.png.svg
- main.html
```

but I still get error
```
raise TemplateNotFound(template)
 jinja2.exceptions.TemplateNotFound: .icons/assets/images/android-chrome-256x256.png.svg
 ERROR: Job failed: exit code 1
```

Where is ".icons".?

Thanks in advance.

**Comment by diba1013** [comment]
I followed your migration guide and it seems to work. Great work on the CSS Variables, took a bit of trial and error, but works fine. Need to fine-tune our CI colors now.

---

@SM-26 

> Where is ".icons"

The icons specified for `icon>logo` are relative to the .icon directory in the theme: 

Please take a look at my migration at https://github.com/retest/docs/tree/feature/mkdocs-material-v5

I specified theme and put them under `theme/.icon/retest/logo.svg` to be in sync with fontawesome and other icon pacs

**Comment by squidfunk** [comment]
Note that icons can only be used as SVGs, as they are inlined into the HTML.

**Comment by wilhelmer** [comment]
So there's `theme.logo`, which can only be set to your own custom-made logo (png, svg, whatever).

And there's `theme.icon.logo`, which can only be set to any of the icons bundled with the theme, e.g., `material/cloud`.

Is that correct? IMHO, that's a bit confusing ...

**Comment by squidfunk** [comment]
The idea was no unify icon configuration, so that maybe later we can add more icons to the `theme.icon` namespace that can be configured. I understand that it may be confusing at first. What would be a better approach? From v4 to v5 it currently is:

- `theme.logo.icon` -> `theme.icon.logo`
- `extra.repo_icon` -> `theme.icon.repo`

**Comment by wilhelmer** [comment]
Maybe:

* `theme.logo.icon` -> `theme.icon.logo`
* `extra.repo_icon` -> `theme.icon.repo`
* `theme.logo` -> `theme.icon.custom_logo`

**Comment by squidfunk** [comment]
`theme.logo` can be any file format

**Comment by wilhelmer** [comment]
And everything in the `theme.icon` namespace should be SVG only? Is that a must?

If yes, I don't know – but the difference between `theme.logo` and `theme.icon.logo` isn't clear just by looking at the names. Maybe `theme.logo` -> `theme.custom_logo`? But the other `.logo` can be customized too, as we just learned ...

**Comment by diba1013** [comment]
I agree that there is an inconsistency between `theme.logo` and `theme.icon.logo` and I [moved](https://github.com/retest/docs/commit/be62b250c1be591ff09f024c2776500530882b1f) the logo to `theme.icon.logo` because it makes more sense to merge these into one namespace. 

But I do not know the code behind that and if the svg locking was done for a reason (e.g. to inline the svgs directly into the html) because otherwise it would have been to complicated or would have resulted in ugly code (e.g. exist check of file, etc).

> But the other .logo can be customized too, as we just learned ...

As to update the documentation, it would be helpful to update the documentation and to make it more clear to what I did. Because what I did&mdash;in retrospect&mdash;seems quite simple, but if you do not have an idea on what I did, it seems complicated. 

To elaborate: My solution (I did not check this) works only, because I have an overwritten theme defined, because it looks relative to a specific `.icons` directory that is only avaialble from the theme. Thus to define `theme.icon.logo` the way I did, you need to overwrite a theme which is a more advanced topic. Whereas `theme.logo` does not care where the icon is placed and therefore a more straightforwards and simple solution.

**Comment by wilhelmer** [comment]
Different topic: 

In the OP, you mention:

* [x]  Improved extensible keyboard handling

Could you elaborate on that, maybe with an example on how to add custom keyboard shortcuts? And maybe also add that to the docs?

**Comment by squidfunk** [comment]
The point is that `theme.icon.logo` is inlined and thus needs to be SVG whereas `theme.logo` is embedded as an `img`. Maybe we should clarify that in the docs. If somebody has an idea how we could better clarify it, I would be happy to merge PRs.

**Comment by squidfunk** [comment]
@wilhelmer the docs on extensibility still need to be written, but in general you can subscribe to the keyboard with `__material.keyboard$.subscribe(console.log)`. Maybe we could add some tutorials before the release. Is there something specific you want to achieve?

**Comment by wilhelmer** [comment]
Let’s say I want to map the **t** key to toggle the visibility of `#myDiv`.

**Comment by jaimeiniesta** [comment]
Congratulations, the upgraded template looks great! I'm evaluating different alternatives to document our project and trying to pick the most accessible template. MkDocs is the best I've found by now.

In case it helps, here's a report on the new template:

https://rocketvalidator.com/s/159ab862-2423-47ca-8772-27db5565cc73/p

And here is one on the previous version:

https://rocketvalidator.com/s/323579ea-0149-4494-92c6-e40184e4df8a/p

There's a new a11y issue on the new template, [ensure that scrollable region has keyboard access](https://rocketvalidator.com/s/159ab862-2423-47ca-8772-27db5565cc73/p/2779116/v?search=Ensure+that+scrollable+region+has+keyboard+access).

**Comment by squidfunk** [comment]
@wilhelmer they keyboard observable emits the following format:

``` js
{
  mode: "global" | "search",
  type: "t",
  claim: [Function]
}
```

The `mode` tells you whether you're in search mode or global mode (search is inactive). The keyboard observable will filter out all keys that have been pressed in conjunction with shift and control keys (or command on macOS) and put the actual key in `type`. `claim` will call `ev.preventDefault` on the original event to "claim" the key. So, with this knowledge you could achieve what you asked for with:

``` js
__material.keyboard$.subscribe(key => {
  if (key.mode === "global" && key.type === "f") {
    var el = document.getElementById("myDiv")
    el.hidden = !el.hidden
    key.claim()
  }
})
```

**Comment by squidfunk** [comment]
@jaimeiniesta thanks for reporting. I can't really make anything out of the report actually, it doesn't say where the actual error is. It only says:

>  Ensure that scrollable region has keyboard access
>  
>   - `<code>`

It would be amazing if you could help fixing the errors in a PR, so we could include it in the next RC.

**Comment by wilhelmer** [comment]
> 
> 
> @wilhelmer the `keyboard$` observable emits the following format:
> 
> ```js
> {
>   mode: "global" | "search",
>   type: "t",
>   claim: [Function]
> }
> ```
> 
> The `mode` tells you whether you're in search mode or global mode (search is inactive). The keyboard observable will filter out all keys that have been pressed in conjunction with shift and control keys (or command on macOS) and put the actual key in `type`. `claim` will call `ev.preventDefault` on the original event to "claim" the key. So, with this knowledge you could achieve what you asked for with:
> 
> ```js
> __material.keyboard$.subscribe(key => {
>   if (key.mode === "global" && key.type === "f") {
>     var el = document.getElementById("myDiv")
>     el.hidden = !el.hidden
>     key.claim()
>   }
> })
> ```

Thanks. Global mode doesn't seem to check for editable elements, so I'd add something like

```js
// Skip editable elements
if ((document.activeElement instanceof HTMLElement && document.activeElement.isContentEditable) || document.activeElement.tagName === "TEXTAREA" || document.activeElement.tagName === "INPUT")
    return;
```

**Comment by squidfunk** [comment]
That's true. Actually, for the included keyboard handlers it does, but I didn't want to impose the checking onto the user when customizing. The checking is done here.

https://github.com/squidfunk/mkdocs-material/blob/c0abaf5692126cb57c8e12dfcfd95a1d37246ab7/src/assets/javascripts/integrations/keyboard/index.ts#L151-L162

... and implemented here:

https://github.com/squidfunk/mkdocs-material/blob/c0abaf5692126cb57c8e12dfcfd95a1d37246ab7/src/assets/javascripts/browser/keyboard/index.ts#L49-L62

**Comment by squidfunk** [comment]
Just released RC 2 which includes fixes for the reported issues. Updated the OP.

**Comment by jaimeiniesta** [comment]
@squidfunk my 2 cents! You can also try with the free Chrome extension for axe on specific pages:

https://www.deque.com/axe/

I'm considering using MkDocs for our own API docs so yes I'll be happy to contribute with PRs to fix the issues I find.

**Comment by squidfunk** [comment]
@jaimeiniesta great, however as discussed, some issues are not fixable by this project as they're more related to the Markdown parser or extensions used. Skimming through the report what I would regard as fixable is:

- Ensures landmarks are unique (probably fixed by `role=...`?)
- All page content must be contained by landmarks (we could set a semantic tag on the hero)
- Elements must have sufficient color contrast (easily fixable by custom CSS if desired)

**Comment by jaimeiniesta** [comment]
@squidfunk thanks, I see, I'll mute the non-fixable issues as they're out of scope and concentrate on what can be fixed now. 👍

**Comment by wilhelmer** [comment]
In 6daac21f6582f2cc474f2fbd199b82b865589b2f, you added `hyphens: auto` to the main text content:

https://github.com/squidfunk/mkdocs-material/blob/6daac21f6582f2cc474f2fbd199b82b865589b2f/src/assets/stylesheets/base/_typeset.scss#L58-L64

Not sure if the majority of users will like this change. I personally don't.

Just checked www.nytimes.com, www.cnn.com and some major German news sites, none of them use hyphenation for their articles.

**Comment by squidfunk** [comment]
@wilhelmer that’s correct. I personally like it because it increases readability. However to disable you can use the following extra CSS:

```
.md-typeset {
  hyphens: initial;
}
```

**Comment by wilhelmer** [comment]
Yes I know, and that's what I did. I just assume users won't be happy with the default. 

I'd say the fact that all major news sites don't have auto hyphens enabled is a strong indication that it does not improve readability. Especially the NY Times is very keen on readability.

**Comment by squidfunk** [comment]
Let’s wait and see how the general feedback is 😉

**Comment by jaimeiniesta** [comment]
I'd prefer to have hyphenation turned off by default as it poses some accessibility problems for people with cognitive impairments or low vision:

https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Hyphenation

**Comment by squidfunk** [comment]
Thanks for your feedback. There is an easy mechanism - using extra CSS. It’s takes 5min to set up 😊

**Comment by ofek** [comment]
What's the hyphen talk about? Screenshot?

**Comment by squidfunk** [comment]
Just check the deploy preview linked in the OP

**Comment by ofek** [comment]
I looked but don't see anything of note...

**Comment by BigIve** [comment]
Just installed the updated mkdocs, I can now no successfully deploy my github repo.  How can I roll back this update?  I'm new to the world of GitHub and am having a very frustrating time of it.  Please advise ASAP.

**Comment by squidfunk** [comment]
@BigIve could you give us some more information on what you’re tying to achieve and what error you’re facing?

**Comment by wilhelmer** [comment]
> I looked but don't see anything of note...

![FA81D1E2-58D3-4E1D-BB6F-3069195B0E58](https://user-images.githubusercontent.com/2234833/76676310-79d0a880-65c2-11ea-908e-a5963a0e078b.jpeg)

**Comment by BigIve** [comment]
Hi,
Ever since I tried to update to the new mkdocs I get the following (see below) when I try to run the 'deploy' command.
Do some further reading it seems that the newer version doesn't support 'string_types' and this seems to be used quite a bit in the mkdocs currently installed.
Unless there is a fix that a 'noob' can do I'd really like to know how to roll back to the earlier version.  As I have a lot of work that I need to do to my documentation repo and can't do anything currently.

**----------------------------------------------------------------------------------------**
PS C:\GitHub\MS-Cloud-Doco> mkdocs gh-deploy
Traceback (most recent call last):
  File "c:\program files (x86)\python38-32\lib\runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files (x86)\python38-32\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Program Files (x86)\Python38-32\Scripts\mkdocs.exe\__main__.py", line 7, in <module>
  File "c:\program files (x86)\python38-32\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "c:\program files (x86)\python38-32\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "c:\program files (x86)\python38-32\lib\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\program files (x86)\python38-32\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\program files (x86)\python38-32\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs\__main__.py", line 178, in gh_deploy_command
    cfg = config.load_config(
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs\config\base.py", line 197, in load_config
    errors, warnings = cfg.validate()
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs\config\base.py", line 107, in validate
    run_failed, run_warnings = self._validate()
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs\config\base.py", line 62, in _validate
    self[key] = config_option.validate(value)
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs\config\config_options.py", line 130, in validate
    return self.run_validation(value)
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs\config\config_options.py", line 591, in run_validation
    plgins[item] = self.load_plugin(item, cfg)
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs\config\config_options.py", line 599, in load_plugin
    Plugin = self.installed_plugins[name].load()
  File "c:\program files (x86)\python38-32\lib\site-packages\pkg_resources\__init__.py", line 2443, in load
    return self.resolve()
  File "c:\program files (x86)\python38-32\lib\site-packages\pkg_resources\__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs_git_revision_date_plugin\plugin.py", line 6, in <module>
    from mkdocs.utils import string_types
ImportError: cannot import name 'string_types' from 'mkdocs.utils' (c:\program files (x86)\python38-32\lib\site-packages\mkdocs\utils\__init__.py)
**---------------------------------------------------------------------------------------**

**Comment by squidfunk** [comment]
@BigIve you have to distinguish two things here - `mkdocs` and `mkdocs-material`. I assume you updated to the release candidate of _Material for MkDocs 5_ which is a theme for `mkdocs` and requires version 1.1 of MkDocs. If you were running version 4.x before, the error may come from stale dependencies due to `pip`s aweful dependency management. My theory is that MkDocs wasn't updated to 1.1. Try to force-reinstall everything (incl. dependencies) with the following command:

```
pip install --upgrade --force-reinstall "mkdocs-material>=5.0.0rc2"
```

**Comment by ofek** [comment]
@wilhelmer Oh my, thanks for the heads-up! We have a few folks with vision impairments so we will definitely disable the hyphens. (I also don't like the hyphens fwiw)

**Comment by squidfunk** [comment]
Could somebody maybe clarify why hyphenation impacts people with vision impairment?

**Comment by squidfunk** [comment]
56bd2b10 reduces the amount of created layers, significantly improving rendering performance.

Before:

<img width="916" alt="Bildschirmfoto 2020-03-14 um 18 05 28" src="https://user-images.githubusercontent.com/932156/76686845-f5604300-661e-11ea-93ae-7b6374315777.png">

After:

<img width="916" alt="Bildschirmfoto 2020-03-14 um 18 05 25" src="https://user-images.githubusercontent.com/932156/76686846-fb562400-661e-11ea-90f7-c74753fb78d4.png">

**Comment by facelessuser** [comment]
Having to follow the broken word to the next line can be hard for people just struggling to read the word. Normally those words don't get broken up.

**Comment by ofek** [comment]
Just asked one such colleague:

> I don't know how it plays with screen readers. So that's a likely issue
  Personally I haven't thought about this because I don't think I've run into this
  But off the top of my head I imagine this being annoying when I'm zoomed in and have to go back to the left side of the screen to finish a word

**Comment by facelessuser** [comment]
Yeah, my wife became aware that she has a genetic eye disorder, and it has impacted here vision in recent years. I just asked her.

**Comment by squidfunk** [comment]
Thanks for the clarification. I find that to be quite astonishing since from my experience nearly all books and magazines use hyphenation.

**Comment by facelessuser** [comment]
> Thanks for the clarification. I find that to be quite astonishing since from my experience nearly all books and magazines use hyphenation.

My wife does audio books now. She didn't use to, but it's gotten harder for her now. Not everyone does things that are friendly to visually impaired.

**Comment by squidfunk** [comment]
I see. Okay then, reverted `hyphens` in bcb8b4a4 then.

**Comment by ofek** [comment]
Hmm, upgrade broke tabs

`5.0.0b2.post1`: https://datadoghq.dev/integrations-core/

`5.0.0rc2`:

![Capture](https://user-images.githubusercontent.com/9677399/76692172-35c4be80-6629-11ea-9eff-f69404aea06e.PNG)

**Comment by facelessuser** [comment]
@ofek RC2 has a deploy preview with working tabs: https://deploy-preview-1486--mkdocs-material-preview.netlify.com/.  Did you configure it correctly? Feature configuration has changed from 4.X.X series.

**Comment by ofek** [comment]
Ah thanks, I had

```
  feature:
    instant: true
    tabs: true
```

vs

```
  features:
    - tabs
    - instant
```

**Comment by BigIve** [comment]
> @BigIve you have to distinguish two things here - `mkdocs` and `mkdocs-material`. I assume you updated to the release candidate of _Material for MkDocs 5_ which is a theme for _MkDocs_ and requires version 1.1 of MkDocs. If you were running version 4.x of `mkdocs-material` before, the error may come from stale dependencies due to `pip`s aweful dependency management. My theory is that `mkdocs` wasn't updated to 1.1 and you're still running 1.0. Try to force-reinstall everything (incl. dependencies) with the following command:
> 
> ```
> pip install --upgrade --force-reinstall "mkdocs-material>=5.0.0rc2"
> ```
> 
> EDIT: I just realized that the error originates in the `mkdocs_git_revision_date_plugin` you're using:
> 
> ```
> ...
>  File "c:\program files (x86)\python38-32\lib\site-packages\mkdocs_git_revision_date_plugin\plugin.py", line 6, in <module>
>     from mkdocs.utils import string_types
> ImportError: cannot import name 'string_types' from 'mkdocs.utils' (c:\program files (x86)\python38-32\lib\site-packages\mkdocs\utils\__init__.py)
> ```
> 
> Thus, the error may be related to the plugin, not this theme. If the reinstallation doesn't help, you may contact the authors of the plugin.

Hi squidfunk, I've tried the command you suggested and am still getting the same error as before.

I've also tried a command I was given to update that date plugin but it doesn't seem to help either.

If I comment out that plugin what functionality will I loose?  Just thinking of doing it as a test.

Thanks for the assistance so far.

**Comment by BigIve** [comment]
I can see an 'issue' on the page for that plugin that seems to acknowledge this error and leads to this page where they mention a fix for the file.
https://github.com/zhaoterryy/mkdocs-git-revision-date-plugin/pull/13/files
However I've tried to re-install and force install the plugin again, and even make a new clone of the repository and the file doesn't seem to be brought down to my PC.  Am very confused.

**Comment by ofek** [comment]
@BigIve I'm using [mkdocs-git-revision-date-localized-plugin](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin), they fixed that https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/issues/8

`pip install "mkdocs-git-revision-date-localized-plugin>=0.4.6"`

**Comment by BigIve** [comment]
@ofek Thanks for the command line.  I ran it and it did seem to take a while to install.  However the file in question hasn't change and the same error as before is still present.

**Comment by ofek** [comment]
@BigIve https://squidfunk.github.io/mkdocs-material/plugins/revision-date/

**Comment by ofek** [comment]
I noticed the new `tabbed` extension changed

`5.0.0b2.post1`:

![Capture](https://user-images.githubusercontent.com/9677399/76695114-c8795380-6651-11ea-97ab-f164d6e8d67f.PNG)

`5.0.0rc2`:

![Capture](https://user-images.githubusercontent.com/9677399/76695257-53a71900-6653-11ea-88cd-2ae239ca7c09.PNG)

That new underline of the current tab is awesome!!! Can we add the outline back though?

**Comment by BigIve** [comment]
> @BigIve https://squidfunk.github.io/mkdocs-material/plugins/revision-date/

@ofek - Thanks for that.  I double checked my plugins list and hadn't correctly changed the one for the date plugin.  So after correcting it some good news.  The above error no longer occurs.

The annoying news is that I now get another error.  As far as I know I haven't changed the lightgallery extension, or is that affected by the change to the mkdocs theme etc?
**---------------------------------------------------------------------------------**
PS C:\GitHub\MS-Cloud-Doco> mkdocs gh-deploy
ERROR   -  Config value: 'markdown_extensions'. Error: Failed loading extension "lightgallery".

Aborted with 1 Configuration Errors!
PS C:\GitHub\MS-Cloud-Doco>
**---------------------------------------------------------------------------------**

**Comment by squidfunk** [comment]
@BigIve I would suggest to move this discussion somewhere else as this is very likely related to your setup as opposed to the latest version of this project. A general support forum like StackOverflow might be a good place to get some help.

**Comment by squidfunk** [comment]
@ofek the new tabs styling will not receive a border by default. The reason for this is that you can now use tabs in other contexts like Admonitions or Details blocks and I’d like to leave some room for additional customization. If you want to add the borders back, a little extra CSS should be sufficient. Just add some padding on `.tabbed-content` and a border on `.tabbed-set` to your liking and you should be good to go.

**Comment by BigIve** [comment]
> @BigIve I would suggest to move this discussion somewhere else as this is very likely related to your setup as opposed to the latest version of this project. A general support forum like StackOverflow might be a good place to get some help.

@squidfunk - Just tried a re-install of the lightbox extension.  And I can once again deploy my mkdocs pages...  yay.

Now that I can once again deploy, I'd like to ask about the 'tabbed' feature that was discussed by @ofek above.  Looking it up it seems really cool.  How do I activate it, Is it part of the new mkdocs that I installed today?  Or is it a seperate plugin?

I've tried following the formatting instructions in the link below but the 'tabs' are not recognised when the page is deployed.

https://deploy-preview-1486--mkdocs-material-preview.netlify.com/extensions/pymdown/#tabbed

**Comment by squidfunk** [comment]
@BigIve you probably forgot to configure it:

``` yml
markdown_extensions:
  - pymdownx.tabbed
```

I can again just stress to [read the docs](https://deploy-preview-1486--mkdocs-material-preview.netlify.com/extensions/pymdown/#configuration) on the features you want to use. Everything is documented in there. If you follow the docs closely, it should work as expected.

**Comment by ofek** [comment]
Ah, that makes sense. Do you have an example CSS block from before? CSS eludes me...

**Comment by squidfunk** [comment]
@okef, well for starters you could try: 

``` css
.tabbed-set {
  border: 1px solid #eee;
}
.tabbed-content {
  padding: 0 1em;
}
```

**Comment by BigIve** [comment]
> @BigIve you probably forgot to configure it:
> 
> ```yaml
> markdown_extensions:
>   - pymdownx.tabbed
> ```
> 
> I can again just stress to [read the docs](https://deploy-preview-1486--mkdocs-material-preview.netlify.com/extensions/pymdown/#configuration) on the features you want to use. Everything is documented in there. If you follow the docs closely, it should work as expected.

@squidfunk Yes, I double checked that I had included it as you showed (and it is).  However, even after doing that I see a 'radio button' each time the formatting for the 'tab' is used.

**Comment by squidfunk** [comment]
@BigIve If it doesn't work, it's almost certainly some problem with your setup. It works for everyone else. I'm sorry but we have to cut the discussion here. Please post your question on StackOverflow or try the official Gitter channel, maybe somebody can help you with your setup. This thread is reserved for problems specific to the RC of version 5.

**Comment by ofek** [comment]
@BigIve Please post your question on StackOverflow (no need to reply here) if this doesn't work but it's a new feature so run: `pip install "pymdown-extensions>=7.0b2"`

**Comment by ofek** [comment]
@squidfunk Thank you!!! FYI changed `padding: 0 1em;` to `padding: 1em 1em;` to get extra space at the bottom

**Comment by ofek** [comment]
It's been a while since 4.x but I recall page refreshes keeping your place. Now F5 takes you to the top (without an anchor selected). Is that intentional?

**Comment by facelessuser** [comment]
@ofek, I'm pretty sure that any issues like that may be because instant is enabled. That's still experimental. I assume if you turn that off, all would probably go back to normal.

**Comment by ofek** [comment]
Yup indeed, that was it! Gonna disable for now

**Comment by squidfunk** [comment]
Scroll restoration is a pretty hard nut to crack. Instant loading already covers a lot of cases but there's still some work to be done.

**Comment by squidfunk** [comment]
@okef - actually, could you open an issue for the problem you described? We should document everything that currently breaks with instant loading in issues.

**Comment by ofek** [comment]
https://github.com/squidfunk/mkdocs-material/issues/1507

**Comment by jaimeiniesta** [comment]
> * Ensures landmarks are unique (probably fixed by `role=...`?)

Thanks @squidfunk - this is the last remaining accessibility issue to fix on my list! 

Trouble is with the `<nav class="md-header-nav md-grid">` element. This triggers this issue:

https://dequeuniversity.com/rules/axe/3.5/landmark-unique?application=rocketvalidator

Should I open a new issue for this?

Also, I'll be happy to provide free accounts on the accessibility checker if you want to check directly - ping me at jaime@rocketvalidator.com and I'll set you up.

**Comment by squidfunk** [comment]
@jaimeiniesta as I understand we just need to add the `role` attribute, so you're encouraged to submit a PR. Thanks for the offer regarding the account, maybe I'll come back to it in the future :-) Until then I'll trust your experience.

**Comment by jaimeiniesta** [comment]
> @jaimeiniesta as I understand we just need to add the `role` attribute

I think a better fix is using `aria-label` to describe the different `nav` elements. I'll submit a PR.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role

**Comment by jaimeiniesta** [comment]
PR ready: https://github.com/squidfunk/mkdocs-material/pull/1516

**Comment by squidfunk** [comment]
Just released RC 3 on PyPI and DockerHub.

**Comment by wilhelmer** [comment]
I think RC3 is good to go. Fire it up 🔥

**Comment by squidfunk** [comment]
@wilhelmer you mean release the final version? Not just yet. My iteration plan is to wait until the end of March if something else pops up. There are so many changes in there, better be sure 😊 if you’re already using the RC there will be no changes anyway, so the upgrade path should be very smooth unless there’s still something to be changed.

**Comment by wilhelmer** [comment]
That’s fine. I‘ll use RC3 in production then.

**Comment by lgeiger** [comment]
Thanks for the great work on version 5! I am really looking forward to the stable release.

I noticed a small glitch when trying out the release candidate: If instant loading is enabled, `extra_javascript` is not properly loaded when switching between pages. `extra_javascript` is only available on first load and disappears after switching to a different page. This is in particular noticeable when using [MathJax](https://squidfunk.github.io/mkdocs-material/extensions/pymdown/#arithmatex-mathjax) or plotting libraries to render content which require additional Javascript as rendered content will disappear when switching to a different page and will only appear again after a full reload.

**Comment by squidfunk** [comment]
@lgeiger this is a limitation of instant loading as we cannot possibly know which scripts must be executed again after a document switch, except those contained in the article (using `script` directly from Markdown). There is however an event to which you can listen on `window` called `DOMContentSwitch` which is emitted after the content was replaced. You could load MathJax once and then trigger the rendering in this event listener.

**Comment by ofek** [comment]
Ah, that's a bummer. Just tried with https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#custom-fences and indeed it's only rendered if it happens to be on the first loaded page (and even gone when navigating back to it)

**Comment by ofek** [comment]
Also, just noticed an annoying thing but unsure if it's a regression: any `h5` has header text in all caps

```
#### test4

##### test5

###### test6
```

![Capture](https://user-images.githubusercontent.com/9677399/77233782-1fb28300-6b80-11ea-94c1-a68341f38893.PNG)


Even the HTML output looks correct too (no caps)

```html
<h4 id=test4>test4<a class=headerlink href=#test4 title="Permanent link">&para;</a></h4> <h5 id=test5>test5<a class=headerlink href=#test5 title="Permanent link">&para;</a></h5> <h6 id=test6>test6<a class=headerlink href=#test6 title="Permanent link">&para;</a></h6>
```

**Comment by squidfunk** [comment]
@ofek the same as for v4, so no regression. You could use some extra CSS to override the default style if you don't like it.

**Comment by squidfunk** [comment]
> Ah, that's a bummer. Just tried with https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#custom-fences and indeed it's only rendered if it happens to be on the first loaded page (and even gone when navigating back to it)

As noted in my previous comment, it's pretty easy to set it up by re-triggering it with `DOMContentSwitch`.

**Comment by ofek** [comment]
Woah yup!

https://github.com/squidfunk/mkdocs-material/blob/dc5d86d52e380fa0ec558a616992143312de0a65/src/assets/stylesheets/base/_typeset.scss#L124-L127

I didn't know CSS could do stuff like that, I wonder how it handles non-ascii...

Since v5 can introduce minor breaking changes, what do you think of removing that capitalization? What was the original intent?

We're in the process of writing an extension (https://github.com/DataDog/mkdocs-click) to auto-generate our CLIs and noticed this for a few subcommands which was _extremely_ unexpected

**Comment by facelessuser** [comment]
> Since v5 can introduce minor breaking changes, what do you think of removing that capitalization? What was the original intent?

I believe the original intent is distinguishing the header level from others. When the levels get even lower, you can't just keep making them smaller, it is no longer practical. In this case, the attempt to differentiate is performed by capitalizing the text. I'm not saying whether I think it is good or bad, simply that it is.

**Comment by squidfunk** [comment]
@facelessuser is right. I’m happy to discuss alternative approaches to distinguish `h4-h6`.

**Comment by ofek** [comment]
I'd take anything that doesn't change which characters are displayed. Maybe a lighter bold?

**Comment by squidfunk** [comment]
If you want you can draft a proposal which we can discuss, otherwise It’s probably best to leave it as it is 😉

**Comment by ofek** [comment]
Done! https://github.com/squidfunk/mkdocs-material/issues/1522

**Comment by ofek** [comment]
I disabled instant loading, but otherwise rc3 is looking great https://datadoghq.dev/integrations-core/

Fantastic work!

**Comment by squidfunk** [comment]
Thanks for all the feedback! I'll try to fix some of the outstanding issues (which are not critical) until the final release. The release will take place on the __7th of April__. I'll perform the merge in the early afternoon of that day.

**Comment by Andre601** [comment]
I would like to point out, that the link about the custom CSS in the [Getting started page](https://deploy-preview-1486--mkdocs-material-preview.netlify.com/getting-started/) - [this one here](https://deploy-preview-1486--mkdocs-material-preview.netlify.com/getting-started/customization.md/#additional-stylesheets) - leads to a broken site, it seems?

Screenshot:
![image](https://user-images.githubusercontent.com/11576465/77677532-20885200-6f90-11ea-8296-e9ae41e1c0f7.png)

**Comment by squidfunk** [comment]
@Andre601 thanks for pointing it out! Fixed in c4820f93, should be shortly available.

**Comment by squidfunk** [comment]
@wilhelmer c7381103 fixes emissions of keyboard events when in editable elements (`input`, etc.), i.e. Material will do the check you did manually in https://github.com/squidfunk/mkdocs-material/issues/1498#issuecomment-598275325 for you.

**Comment by wilhelmer** [comment]
Thanks for the fix and the release announcement. 

If you had to publish a documentation project on April 1st (no joke) and wanted to use Material v5, would you use RC3 or the latest version from the `rxjs-typescript` branch?

**Comment by squidfunk** [comment]
I think it's probably safest to use RC3 for now, although it has some very minor bugs.

**Comment by lazka** [comment]
Using 5.0.0rc3 the build fails when using social links:

```python
Traceback (most recent call last):
  File "/home/lazka/.local/bin/mkdocs", line 10, in <module>
    sys.exit(cli())
  File "/home/lazka/.local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/lazka/.local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/lazka/.local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/lazka/.local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/lazka/.local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/lazka/.local/lib/python3.8/site-packages/mkdocs/__main__.py", line 159, in build_command
    build.build(config.load_config(**kwargs), dirty=not clean)
  File "/home/lazka/.local/lib/python3.8/site-packages/mkdocs/commands/build.py", line 288, in build
    _build_theme_template(template, env, files, config, nav)
  File "/home/lazka/.local/lib/python3.8/site-packages/mkdocs/commands/build.py", line 114, in _build_theme_template
    output = _build_template(template_name, template, files, config, nav)
  File "/home/lazka/.local/lib/python3.8/site-packages/mkdocs/commands/build.py", line 93, in _build_template
    output = template.render(context)
  File "/home/lazka/.local/lib/python3.8/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/home/lazka/.local/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/home/lazka/.local/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/lazka/.local/lib/python3.8/site-packages/material/404.html", line 4, in top-level template code
    {% extends "base.html" %}
  File "/home/lazka/.local/lib/python3.8/site-packages/material/base.html", line 172, in top-level template code
    {% block footer %}
  File "/home/lazka/.local/lib/python3.8/site-packages/material/base.html", line 173, in block "footer"
    {% include "partials/footer.html" %}
  File "/home/lazka/.local/lib/python3.8/site-packages/material/partials/footer.html", line 59, in top-level template code
    {% include "partials/social.html" %}
  File "/home/lazka/.local/lib/python3.8/site-packages/material/partials/social.html", line 10, in top-level template code
    {% include ".icons/" ~ social.icon ~ ".svg" %}
  File "/home/lazka/.local/lib/python3.8/site-packages/jinja2/loaders.py", line 199, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: .icons/.svg
```

The error starts once I add the social config from the docs:

```yaml
extra:
  social:
    - type: 'github'
      link: 'https://github.com/squidfunk'
    - type: 'twitter'
      link: 'https://twitter.com/squidfunk'
    - type: 'linkedin'
      link: 'https://www.linkedin.com/in/squidfunk'
```

**Comment by squidfunk** [comment]
`type` was renamed to `icon`, as stated in the migration guide.

**Comment by lazka** [comment]
Ah, thanks.

**Comment by Andre601** [comment]
A suggestion for the [icons section](https://deploy-preview-1486--mkdocs-material-preview.netlify.com/getting-started/#icons) of the new guide:
Maybe mention how the names of the different packs are?
I mean you mention material and fontawesome, but what would be the "prefix" for octicons? Just octicons, or something else?

And not sure how important this would actually be for this theme, but perhaps mention that yiz can alter the edit-uri by providing `edit_uri` in the mkdocs.yml? I myself had a hard time finding this anywhere and only found it through luck... But I guess this is more the job of the actual mkdocs documentation?

**Comment by squidfunk** [comment]
I‘m open to PRs improving the docs, especially on migration. It’s even better coming from actual users than me. 

However, as you would have to loon up the name of the icon anyway, any idea was that stating the name of the folder where those icons reside was probably enough. We can’t list all of the 5k icons, obviously.

**Comment by Andre601** [comment]
> We can’t list all of the 5k icons, obviously.

My point was, that it should at least mention the how I called it "prefix" (i.e. fontawesome) as it's not 100% sure for people where to look for the icons.

And a unrelated question, but still about the new theme version:
Did something change in terms of custom CSS?
It seems that the codehilite colours for code blocks no longer seems to apply properly as they did for the dark theme CSS I used with the stable release.

I use the [dark Theme CSS file from henrywhitake3](https://github.com/henrywhitaker3/mkdocs-material-dark-theme)

**Comment by squidfunk** [comment]
Unfortunately we cannot provide support for downstream projects. Some styling (i.e. selectors) changed for code highlighting, so yes, it might be related to that. It’s probably best to create an issue on the respective repository.

**Comment by Andre601** [comment]
found the problem: codehilite changed to highlight it seems.

**Comment by squidfunk** [comment]
Both highlighting extensions are supported as in 4.x, soon change in that front. Depends on whether you use `pymdownx.highlight` or `codehilite` in `mkdocs.yml`.

**Comment by Andre601** [comment]
I use the codehilite one, which makes this even more interesting/weird.
I'll open an issue on the repo of the dark theme to suggest an alternative css file supporting the highlight option.

**Comment by squidfunk** [comment]
@facelessuser any ideas what could be causing this?

**Comment by facelessuser** [comment]
@squidfunk, just curious, I noticed that the included Material icons are a mix of official and contributed, I assume it from this collection: https://materialdesignicons.com/. I noticed that some icons have IDs attached to them, for instance I think some of the brand icons in Material (GitHub for instance). Depending on how they are inlined, it could much with other ids. When you insert them via Jinja, do you strip the IDs? I figure when they are set as `background` or `mask-image` it shouldn't be a problem. Just something I noticed and am curious about.

**Comment by squidfunk** [comment]
On a second note: one of the next minor version, most probably 5.1 will include a native dark mode.

**Comment by squidfunk** [comment]
@facelessuser the icons are extracted from the `material-design-icons-svg` package which seems to scrape the official site. I included the IDs at some point but removed that again from the Webpack copying logic.

**Comment by facelessuser** [comment]
@squidfunk @Andre601 SuperFences and InlineHilte use `pymdownx.highlight` to handle code highlighting, not Codehilite. For a long time they mirrored the settings if you were using CodeHilite, but it still always used `pymdownx.highlight`.

So in Material 4.X, if you used codehilite, `pymdownx.highlgiht` would copy over the class name used by codehilite, which is `codehilite`. But in Material 5.X (which uses pymdownx 7.0 prerelease) does not. `pymdownx.highlight` uses `highlight` by default, but you can configure it in `pymdownx.highlight`, it should work along side codehilite.

**Comment by squidfunk** [comment]
Thanks for the explanation!

**Comment by facelessuser** [comment]
> @facelessuser the icons are extracted from the material-design-icons-svg package which seems to scrape the official site. I included the IDs at some point but removed that again from the Webpack copying logic.

Cool, so no IDs, good to know :+1:.

**Comment by facelessuser** [comment]
The big thing you gain from using CodeHilite is indented code block headers with shebangs, and `markdown.extensions.fenced_code` uses CodeHilite for its highlighting.

```
    #!python
    import sys
```

But if you are using `pymdownx.superfences` you don't really need it:

````
```python
import sys
```
````

I find no use for CodeHilite, so I no longer try to support anything related to it. I only mirrored the options for a while to transition away from using CodeHilite (which SuperFences used to use a long long time ago).

**Comment by facelessuser** [comment]
Last post I swear 🙂 . The super short answer:

The referenced dark theme only uses the class `codehilite`:  https://github.com/henrywhitaker3/mkdocs-material-dark-theme/blob/master/site/assets/css/dark-theme-system.css#L153. So you need to use your own with updated classes (`highlight`) or force `pymdownx.highlight` to use the class name `codehilite`.

**Comment by squidfunk** [comment]
... or wait for 5.1, which should shortly follow after the initial release 😉 probably towards the end of April.

**Comment by squidfunk** [comment]
I decided to release a last release candidate, as there have been a lot of fixes lately. The final release date will be April 7.

**Comment by wilhelmer** [comment]
Printing is not working properly for me in Firefox v74. 

Steps to reproduce:

1. Set up a minimal project with one fairly long topic (>1 page when printed).
2. Build.
3. Open topic in Firefox 74.
4. File > Print preview.

Result: The print preview shows only one page.

Expected result: The print preview should show all pages needed to print the topic.

Works in Chrome, but not in FF.

**Comment by wilhelmer** [comment]
Also, `__material` is no longer defined in RC4? Can't access it from the console either. Did I overlook that change?

**Comment by squidfunk** [comment]
`__material` was renamed to `app`. Does the print problem happen with Firefox and v4?

**Comment by wilhelmer** [comment]
Thanks. No, does not happen with FF and v4.

**Comment by squidfunk** [comment]
Confirmed. Could you create a new issue (or PR?)? I guess it has to do with `overflow: hidden` and `flex` which should be easy to remove for print styles on `body`.

**Comment by lazka** [comment]
Instant loading bug (Firefox and Chromium):

* Click on link to trigger instant loading
* History back
* Try to click on the same link again
* Actual: Nothing happens
* Expected: The link works

**Comment by squidfunk** [comment]
@lazka thanks for reporting! Could you create a new issue, so we can keep track?

**Comment by jaimeiniesta** [comment]
Awesome! Just deployed our [docs](https://docs.rocketvalidator.com) and [no html/a11y issues](https://rocketvalidator.com/s/8cc044ff-180d-4819-9316-070109491a14/p) were detected this time - except for the muted ones we already discussed. 💪

**Comment by Andre601** [comment]
I'm not entirely sure if this would be for here, or is better suited for a markdown-expansion like the ones of pymdown, but how about a implementation, to easy add any icon of material, fontawesome or github?
I like that you can use them for social links and other specific places, but I think me and other people would love it, if there was a way to directly put those icons into the text, without requiring this entire `<i class="blablabla">` html stuff (keep it very "markdown like" if you will)

A syntax I think would be an idea is perhaps `|prefix:icon|`?
Like I said, not sure if this is best suited here or better of for a markdown-extension.

**Comment by facelessuser** [comment]
@Andre601 some ideas for this are already being floated: https://github.com/squidfunk/mkdocs-material/pull/1475.

**Comment by Andre601** [comment]
A bit unrelated question, but would I just need to use [this part here](https://github.com/squidfunk/mkdocs-material/blob/master/.github/workflows/ci.yml#L64-L105) if I want to make a GitHub Action automatically update a page, when a PR was merged on the repo containing the docs?
I am aware that [there is a GitHub Action](https://github.com/marketplace/actions/deploy-mkdocs) more or less suited for this, but I'm quite sure it doesn't use Material 5.0.0 RC4 and perhaps won't even really support the PyMdown 7 extensions....

**Comment by squidfunk** [comment]
You could use the workflow as a starting point. From what I read from the MkDocs action, however, it seems to assume a `requirements.txt` file that is installed, so you could probably specify the versions you want. Haven't tested it, though.

**Comment by Andre601** [comment]
Is there any way I can force MkDocs/Material to put a single entry in the nav-list to the tabs at the top?
When I f.e. only have `- Contribute: contribute.md` does it not work.
When I, however, set the following does it work (But it is a bit annoying)
```yml
nav:
- Contributing:
  - Contributing: contribute.md
```

**Comment by facelessuser** [comment]
I also deploy docs with just this: https://github.com/facelessuser/pymdown-extensions/blob/master/.github/workflows/deploy.yml#L10. I found it just easier to manually do it. Granted, I didn't need all the matrix stuff as I'm only running it on one Python, but I copied and pasted it from other workflows I was doing, so 🤷‍♂ .

**Comment by ofek** [comment]
Yeah we're doing it manually too https://github.com/DataDog/integrations-core/blob/hack-a-doc/datadog_checks_dev/datadog_checks/dev/tooling/commands/docs/push.py

**Comment by facelessuser** [comment]
I'm fine with using actions when it is nice a solid and does exactly what I want, but when it doesn't, and it is pretty easy to just manually do it, I usually just do it myself.

**Comment by squidfunk** [comment]
@Andre601 from [the docs](https://squidfunk.github.io/mkdocs-material/getting-started/#tabs):

> When tabs are enabled, top-level sections will be rendered in an additional layer directly below the header. The navigation on the left side will only include the pages contained within the selected section. Furthermore, top-level pages defined inside your project's mkdocs.yml will be grouped under the first tab which will receive the title of the first page.

Maybe look at `mkdocs.yml` of the project itself, maybe it's clearer then:

https://github.com/squidfunk/mkdocs-material/blob/728fef6ec8259e910500885047911c18bae145da/mkdocs.yml#L114-L134

**Comment by Andre601** [comment]
So tl;dr it's not possible to have it as a separate tab at the top then. Got it.

**Comment by squidfunk** [comment]
You could always use theme extension and override `partials/tabs.html` if you'd like more control.

**Comment by fkorotkov** [comment]
@Andre601 you can take a look at my solution for the issues that I dod for https://cirrus-ci.org to have single pages on top of the page. Here are [two partials](https://github.com/cirruslabs/cirrus-ci-docs/tree/master/theme/partials) you can take a look at.

**Comment by Andre601** [comment]
I'm not really into implementing own HTML files, which is the entire reason I use MkDocs in the first place: To not have to deal with a lot of HTML, CSS and other configuration.

**Comment by fkorotkov** [comment]
There are minor changes if you'll diff against the original `nav-item`. But yeah, I feel you 😅

**Comment by Andre601** [comment]
A bit sad, that there isn't really a main chat for common mkdocs/material questions.
I know that there's a chat on gitter, but it's more about Material and not MkDocs according to the welcome message there, and Stackoverflow is way to unreliable and slow for some small, common questions.
I also don't want to annoy you guys here constantly with such unrelated questions (which also triggers notification for everyone watching this issue 👀)

**Comment by ofek** [comment]
This looks promising https://github.com/zeit/now/discussions/3874

**Comment by squidfunk** [comment]
> To not have to deal with a lot of HTML, CSS and other configuration.

Unfortunately, it’s impossible to provide a theme that fits everyone’s use case. I understand that you don’t want to go down that path, but you also have to understand that our time is limited, so we have to cut configuration at some point for customization. Funnily, the more configuration options this theme gets, the more users want even more 😅

> A bit sad, that there isn't really a main chat for common mkdocs/material questions.

I think the main problem is that something like this would result in endless requests for customizations, and I think nobody (including me) wants to do all of this for free.

**Comment by Andre601** [comment]
> I think the main problem is that something like this would result in endless requests for customizations, and I think nobody (including me) wants to do all of this for free.

I don't generally mean about new features here, but rather that there isn't a common chat/place where you could ask all kinds of questions, which aren't always about material or even MkDocs (e.g. how to implement something like netlifly to build previews from pull requests)

**Comment by squidfunk** [comment]
I think that's pretty much what StackOverflow is for. Additionally there might be Slack or Discord channels for this topic.

**Comment by wilhelmer** [comment]
> 
> 
> I think it's probably safest to use RC3 for now, although it has some very minor bugs. Nevertheless, I think publishing something on April 1 is never a good idea 😉That's also why I moved the release date to April 7, as Tuesdays should be the best weekday for releases and March 31 might already be April 1 in some timezones.

We went live with RC4 yesterday! 🍾 https://docs.baslerweb.com

Good thing about launching on April 1st: If there are major bugs, you can always say haha first of April, was just a joke, we'll release the real version tomorrow ...

**Comment by rohancragg** [comment]
When trying out search tokenisation, I found I had to put the seaprator delcaration in quotes
(per: https://deploy-preview-1486--mkdocs-material-preview.netlify.com/plugins/search/#tokenization )

like this

```jinja
plugins:
  - search:
      separator: '[\s\-\.]+'
```

**Comment by squidfunk** [comment]
@rohancragg thanks for reporting! Fixed the docs in 2cc06859.

**Comment by rohancragg** [comment]
I found that I had to make the following change to my PIP requirements.txt:

```
pymdown-extensions>=7.0b2
mkdocs-material>=5.0.0rc4
```

**Comment by squidfunk** [comment]
RC 4 has the correct version of Pymdown as a dependency, so no need to specify it. A compatible version is always included with the theme.

**Comment by rohancragg** [comment]
> RC 4 has the correct version of Pymdown as a dependency, so no need to specify it. A compatible version is always included with the theme.

OK, thanks, it's probably just because I'd been specifying it and didn't need to

**Comment by rohancragg** [comment]
I had to remove the reference to   `- pymdownx.superfences`  in my `mkdocs.yml`

You might want to add that to the upgrade steps.

**Comment by squidfunk** [comment]
`pymdownx.superfences` should continue to work.

**Comment by Andre601** [comment]
> RC 4 has the correct version of Pymdown as a dependency, so no need to specify it. A compatible version is always included with the theme.

Good to know.

**Comment by squidfunk** [comment]
It's also mentioned in the new docs – I have removed all installation instructions for packages that are bundled anyway (even MkDocs!) 😀For Pymdown it makes sense to let the theme select the supported version. Also, @facelessuser, the author of Pymdown Extensions, is very responsive, so the package will almost likely almost be up-to-date.

**Comment by facelessuser** [comment]
yup, SuperFences 100% works. All of it works with Material, though not everything pymdown-extensions does is directly supported with styling, but most like 99%.

**Comment by rohancragg** [comment]
> `pymdownx.superfences` should continue to work.

If I don't remove it get this warning on `mkdocs build`:

```
[~]\scoop\apps\python\3.8.2\lib\site-packages\pymdownx\superfences.py:634: PymdownxDeprecationWarning:
The tab option in SuperFences has been deprecated in favor of the general purpose 'pymdownx.tabbed' extension.
While you can continue to use this feature for now, it will be removed in the future.
Also be mindful of the class changes, if you require old style classes, please enable the 'legacy_tab_classes' option.

  warnings.warn(MSG_TAB_DEPRECATION, PymdownxDeprecationWarning)
```

is this perhaps because I've enabled this:

```yaml
features:
    - tabs
    - instant
```

**Comment by facelessuser** [comment]
It says the [tab option] (https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#tabbed-fences) has been deprecated in SuperFences, not that SuperFences is deprecated. Moving forward, people should use the general Tabbed extension to do tabbed code, but during the transition period, you can still use SuperFences' tab option, but it'll warn you that one day it won't work, so it is probably better to use the general extension some time soon to prevent bring caught unawares.

**Comment by rohancragg** [comment]
@facelessuser - thanks, yes, I've now removed all use of tabbed fences and the warning goes away!

**Comment by Andre601** [comment]
Is it intentional, that Material Theme now omits the link to MkDocs rather than adding its own link to the footer? (`made with <Material for MkDocs>` instead of `powered by <MkDocs> and <Material for MkDocs>` (`<>` mean that those words have links))

**Comment by squidfunk** [comment]
Yes. That's an intentional change. It reduces the clutter in the footer. Before, MkDocs was the first link, so people go to the MkDocs page with no Material theme whatsoever. Since Material "brings its own MkDocs" (as a requirement), there's no need to link to MkDocs itself.

I thought about making it uppercase, you're probably right.

**Comment by Andre601** [comment]
In response to the footer (again):
Wouldn't it make a bit more sense to change it to something like "Design provided by [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/)"?

Because it looks a bit weird when you have a text like the following one and there is a second "Made with ..." text
![footer](https://cdn.discordapp.com/attachments/534685730614673428/696466451191562240/unknown.png)

But I assume I could alter the footer text somehow (through override?)

**Comment by squidfunk** [comment]
I assume you added the „made with ❤️...“ yourself, so yeah it doubles in your case but normally there’s only the copyright notice and the theme notice.

I’m open to suggestions regarding the wording, but I want to get rid of „powered by“ as that is the wrong message after thinking about it again. And „built with“ is too technical.

**Comment by ofek** [comment]
@Andre601 https://github.com/facelessuser/pymdown-extensions/blob/1cec43f146cdd41dc9515be03c0bc105e85ae953/mkdocs.yml#L6-L9

**Comment by squidfunk** [comment]
... also the theme provides much more than just the design. The whole UX.

**Comment by Andre601** [comment]
> @Andre601
> 
> https://github.com/facelessuser/pymdown-extensions/blob/1cec43f146cdd41dc9515be03c0bc105e85ae953/mkdocs.yml#L6-L9

What are you saying me with this?

**Comment by ofek** [comment]
You were looking for an example of an override, no?

**Comment by Andre601** [comment]
> You were looking for an example of an override, no?

This doesn't override the "made with ..." text, just fyi
![image](https://user-images.githubusercontent.com/11576465/78510608-d032a080-7796-11ea-912f-5a9f1fff20e0.png)

**Comment by ofek** [comment]
Ah, my bad

**Comment by squidfunk** [comment]
@Andre601 of course you can adjust it using theme extension – just bring your own `partials/footer.html` 😊

**Comment by Andre601** [comment]
> @Andre601 of course you can adjust it using theme extension – just bring your own `partials/footer.html` 😊

Alright. I'll need to look into what I have to actually alter, to not mess stuff up... Sure will be fun... I-I hope

**Comment by squidfunk** [comment]
@Andre601 https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme - the docs describe exactly the overriding of the footer 😉

**Comment by Andre601** [comment]
👍 
![image](https://user-images.githubusercontent.com/11576465/78512071-dd08c180-77a1-11ea-8f5e-f6fdf0ef7d00.png)

https://github.com/purrbot-site/Docs/blob/feature/footer-customisation/override/main.html

**Comment by facelessuser** [comment]
@Andre601 Think you meant `Built` instead of `Build`?

**Comment by Andre601** [comment]
> @Andre601 Think you meant `Built` instead of `Build`?

Not a pro with English grammar...
![image](https://user-images.githubusercontent.com/11576465/78512611-72a65000-77a6-11ea-82ec-8eeef99c020b.png)

**Comment by ofek** [comment]
@facelessuser For the `instant` feature and loading of extra JS described here https://github.com/squidfunk/mkdocs-material/issues/1498#issuecomment-601020251

Will v7 automatically handle that for the new UML library?

- https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#custom-fences
- https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#uml-diagram-example

Mermaid looks absolutely beautiful btw 😄

**Comment by facelessuser** [comment]
> @facelessuser For the instant feature and loading of extra JS described here #1498 (comment)

@ofek v7 of pymdownx doesn't provide any JS automatically.  It didn't automatically provide JS for the old UML either. UML examples are exactly that, examples of how SuperFences can be expanded.

With that said, what I have I believe works with instant (last I checked as I don't currently run instant). I added the "switch" event in my own "extra JS": https://github.com/facelessuser/pymdown-extensions/blob/master/docs/src/js/extra.js#L7.

I don't have a solution yet for MathJax loading in the traditional way, but as I only need it on one page, I currently include it at the end of my Arithmatex page, and it seems to work fine. As I'm not running instant by default, I haven't looked into getting it working in the more traditional way that people would normally load it. 

> Mermaid looks absolutely beautiful btw 😄

Thanks! I really wanted it to kind of go with the overall theme I had going on in the docs, I think it turned out pretty good 🙂.

Mermaid, annoyingly, no longer release CSS, and instead opt to inline the CSS in the exported SVG. So, I include the source in the build process, import their SCSS, and build my own with my own colors and some style overrides. Then we configure Mermaid to not load an inline CSS, and then include our own 😉 .

I think Mermaid provides way more, and seems quite active. The old Sequence Diagram that I used to use works, but I think is pretty much dead, and the latest isn't really compatible with the latest Flowchart.js.

Anyways, I find Mermaid more powerful despite its quirks here and there, and I've seen a number of people preferred it over the old suggestions, so I thought I'd run with it.

**Comment by facelessuser** [comment]
By the way, I use a custom loader for Mermaid to overcome some issues with loading diagrams from hidden tabs and details elements. I mention that in the docs as well.

**Comment by wilhelmer** [comment]
In v4, if you started searching before `search_index.json` was loaded, search results were displayed as soon as the index had been loaded. 

In v5, if you start searching before the index is loaded, the search displays "No matching results" and keeps displaying that message even after the index has been loaded. 

For example, if you type "test" and wait until the search index is loaded, Material keeps displaying "No matching results" until you press a key. You must change the search term to, e.g., "tes" and then back to "test" to get your search results.

Steps to reproduce:

1. Visit https://docs.baslerweb.com
2. Enable DevTools
3. Disable cache and throttle to Fast 3G or slower
4. Page refresh
5. Start searching for "test" immediately

**Comment by squidfunk** [comment]
@wilhelmer could you open an issue? I don’t consider it blocking for the release tomorrow but we should definitely fix it.

**Comment by wilhelmer** [comment]
Yep, will do. And agree, please don't block the release, as this will only affect projects with large search indexes (like ours).

**Comment by squidfunk** [comment]
It’s actually related to the issue that proposes the introduction of a loader during index time, #1439.

**Comment by wilhelmer** [comment]
Should I add a comment there or would you rather keep it as two separate issues?

**Comment by squidfunk** [comment]
I‘d say the open issue is sufficient, so add a comment there regarding the behavior you described.

**Comment by squidfunk** [comment]
**It's done! 🎉🎉🎉 Material for MkDocs 5 is out!**

A big **THANK YOU** to all of you guys testing, reporting bugs, improving it! Without your extensive testing, we wouldn't have such a stable release!

**Comment by ofek** [comment]
Thank you!!!

**Comment by lgeiger** [comment]
Congrats on v5!

> this is a limitation of instant loading as we cannot possibly know which scripts must be executed again after a document switch, except those contained in the article (using script directly from Markdown). There is however an event to which you can listen on window called DOMContentSwitch which is emitted after the content was replaced. You could load MathJax once and then trigger the rendering in this event listener.

Thanks for the help. This solution works very well.
For MathJax 3 simply adding the following few lines makes it work smoothly with instant loading:
```js
// Re-render MathJax on document switch (instant loading, custom event)
document.addEventListener("DOMContentSwitch", function() {
  MathJax.typesetPromise();
});
```
